vendors/swedbank-pay-core-library/src/SwedbankPay/Core/Library/Methods/Invoice.php

```diff
@@ -0,0 +1,374 @@
+        } catch (\Exception $e) {
+            $this->log(LogLevel::DEBUG, sprintf('%s::%s: API Exception: %s', __CLASS__, __METHOD__, $e->getMessage()));
+
+            throw new Exception($e->getMessage());
+        }
+
+        return $result;
+    }
+
+    /**
+     * Capture Invoice.
+     *
+     * @param mixed $orderId
+     * @param int|float $amount
+     * @param int|float $vatAmount
+     * @param array $items
+     *
+     * @return Response
+     * @throws Exception
+     */
+    public function captureInvoice($orderId, $amount = null, $vatAmount = 0, array $items = [])
+    {
+        /** @var Order $order */
+        $order = $this->getOrder($orderId);
+
+        if (!$amount) {
+            $amount = $order->getAmount();
+            $vatAmount = $order->getVatAmount();
+        }
+
+        if (!$this->canCapture($orderId, $amount)) {
+            throw new Exception('Capturing is not available.');
+        }
+
+        $paymentId = $order->getPaymentId();
+        if (empty($paymentId)) {
+            throw new Exception('Unable to get payment ID');
+        }
+
+        $href = $this->fetchPaymentInfo($paymentId)->getOperationByRel('create-capture');
+        if (empty($href)) {
+            throw new Exception('Capture is unavailable');
+        }
+
+        $params = [
+            'transaction' => [
+                'activity' => 'FinancingConsumer',
+                'amount' => (int)bcmul(100, $amount),
+                'vatAmount' => (int)bcmul(100, $vatAmount),
+                'description' => sprintf('Capture for Order #%s', $order->getOrderId()),
+                'payeeReference' => $this->generatePayeeReference($orderId)
+            ],
+            'itemDescriptions' => $items
+        ];
+
+        $result = $this->request('POST', $href, $params);
+
+        // Save transaction
+        $transaction = $result['capture']['transaction'];
+        $this->saveTransaction($orderId, $transaction);
+
+        switch ($transaction['state']) {
+            case 'Completed':
+                $this->updateOrderStatus(OrderInterface::STATUS_CAPTURED, 'Transaction is captured.');
+                break;
+            case 'Initialized':
+                $this->updateOrderStatus(OrderInterface::STATUS_AUTHORIZED,
+                    sprintf('Transaction capture status: %s.', $transaction['state']));
+                break;
+            case 'Failed':
+                $message = isset($transaction['failedReason']) ? $transaction['failedReason'] : 'Capture is failed.';
+                throw new Exception($message);
+            default:
+                throw new Exception('Capture is failed.');
+        }
+
+        return $result;
+    }
+
+    /**
+     * Cancel Invoice.
+     *
+     * @param mixed $orderId
+     * @param int|float|null $amount
+     * @param int|float $vatAmount
+     *
+     * @return Response
+     * @throws Exception
+     */
+    public function cancelInvoice($orderId, $amount = null, $vatAmount = 0)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The method cancelInvoice() has a Cyclomatic Complexity of 10. The configured cyclomatic complexity threshold is 10.](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-woocommerce-payments/pullRequest?prid=5340032)